extensions/wsc/src/main/java/org/mule/extension/ws/internal/introspection/RequestBodyGenerator.java

```diff
@@ -0,0 +1,113 @@
+   */
+  public String generateRequest(WscConnection connection, String operation) {
+
+    BindingOperation bindingOperation = connection.getWsdlIntrospecter().getBindingOperation(operation);
+    List<String> soapBodyParts = getSoapBodyParts(bindingOperation);
+    Message message = bindingOperation.getOperation().getInput().getMessage();
+    Optional<Part> part = getSinglePart(soapBodyParts, message);
+
+    // Checks that the message has a single part with at least one element defined.
+    if (part.isPresent() && part.get().getElementName() != null) {
+      XmlTypeLoader loader = new XmlTypeLoader(connection.getWsdlIntrospecter().getSchemas());
+      Optional<MetadataType> loadedType = loader.load(part.get().getElementName().toString());
+      if (loadedType.isPresent()) {
+        MetadataType metadataType = loadedType.get();
+        if (metadataType instanceof ObjectType) {
+          ObjectType type = (ObjectType) ((ObjectType) metadataType).getFields().iterator().next().getValue();
+          if (type.getFields().isEmpty()) {
+            QName element = part.get().getElementName();
+            return format(NO_PARAMS_SOAP_BODY_CALL_MASK, element.getLocalPart(), element.getNamespaceURI());
+          }
+        }
+      }
+    }
+    throw new WscException(format("No payload was provided for the operation [%s] and a default one cannot be built", operation));
+  }
+
+  /**
+   * Finds the part of the input message that must be used in the SOAP body, if the operation requires only one part.
+   *
+   * @param soapBodyParts the body parts discovered in the binding type
+   * @param inputMessage  the input {@link Message} of the operation.
+   */
+  private Optional<Part> getSinglePart(List<String> soapBodyParts, Message inputMessage) {
+    if (soapBodyParts.isEmpty()) {
+      Map parts = inputMessage.getParts();
+      if (parts.size() == 1) {
+        return Optional.ofNullable((Part) parts.values().iterator().next());
+      }
+    } else {
+      if (soapBodyParts.size() == 1) {
+        String partName = soapBodyParts.get(0);
+        return Optional.ofNullable(inputMessage.getPart(partName));
+      }
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Retrieves the list of SOAP body parts of a binding operation if defined.
+   *
+   * @param bindingOperation the binding operation that we want to get the SOAP body parts from.
+   */
+  public static List<String> getSoapBodyParts(BindingOperation bindingOperation) {
+    List elements = bindingOperation.getBindingInput().getExtensibilityElements();
+    List result = null;
+    for (Object element : elements) {
+      if (element instanceof SOAPBody) {
+        result = ((SOAPBody) element).getParts();
+        break;
+      }
+      if (element instanceof SOAP12Body) {
+        result = ((SOAP12Body) element).getParts();
+        break;
+      }
+    }
+    return result != null ? result : emptyList();
```

Comment:
in the old `getSoapBodyParts`, it used a null return to represent "the absence of SOAP body", are we handling that differently now or we lost a case? 
(is that what `return Optional.ofNullable((Part) parts.values().iterator().next());` is for in `getSinglePart`?)